courses/koopa_troopa_beach/course_data.inc.c

```diff
@@ -8538,7 +8538,7 @@ Gfx d_course_koopa_troopa_beach_dl_16990[] =
     gsSPEndDisplayList(),
 };
 
-// 0x60169b0
+//! 0x60169b0, @todo convert to Gfx
```

Comment:
This and `d_course_yoshi_valley_unk5` aren't Gfx lists. They are their own, arcane thing. Its something related to matrix manipulation, seemingly related to 3d animation (segualls flapping their wings, the flags in YV flapping in the wind, the penguins running around, etc).

Reply:
so why in code they are cast like Gfx ?

Reply:
Because for some objects (but not all them) that member is a Gfx list. But for others its this weird, u32 list. There is a real possibility that there are several different object types in the same way there's lots of actor types. Its just that no one has gone through and identified what all the various objects really need/use.

As such, any Gfx* type casting going on is an artifact of decompilation.

Reply:
and use as an gfx I mean by that is they are use like Gfx so how even it's skeleton/animate model are use in this way ?

Reply:
That my memory serves, `code_80004740` is responsible for handling all the model animation stuff, so if you want to know exactly how its used, you'll have to go digging there.

You'll want to read through the comment at the top of `code_80004740.h`, its my best effort to explain what's going on there. I'm way out of my expertise on this one so that comment may or may not be useful

Reply:
ok so I remove it or replace by just "more research are needed to know the type"

Reply:
And I have no idea if this stuff is similar to SM64 or not. I have looked at SM64's model animation stuff and while it doesn't look similar to me, but I'm really not really knowledgeable enough to say that with certainty.

Reply:
I will just remove the todo and see later code_80004740

Reply:
If I had to guess. The animation data are frame data. Perhaps something like "frame, some other stuff like x,y,z"

Reply:
I think I will search that tomorrow to try understand